### PR TITLE
Clearer project build

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

**Comment by sbrunner** [comment]
@arnaud-morvan 

When I do make help I get:
```
Possible targets:
awk: line 1: syntax error at or near :
Makefile:6: recipe for target 'help' failed
make: *** [help] Error 2
```
I suspect that this kind of line will return an error:
https://github.com/camptocamp/c2cgeoportal/pull/5256/files#diff-b67911656ef5d18c4ae36cb6741b7965R12

Do you have an idea on how to fix it?

File: Makefile [hunk 1]
```diff
@@ -1,310 +1,78 @@
...
+	@echo
+	@echo "Possible targets:"
+	@grep -Eh '^[a-zA-Z_-]+:.?## .$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "	%-20s%s\n", $$1, $$2}'docker-build:
 
 .PHONY: build
-build: \
-	geoportal/c2cgeoportal_geoportal/scaffolds/update/CONST_create_template/ \
-	$(APPS_FILES_ALT) \
-	$(MO_FILES)
+build: \ ## Build all docker images
```
Comment (arnaud-morvan) [inline Makefile]: Put a line just for the help message
```
.PHONY: build
build: ## Build all docker images
build: \
    ...
```

Comment (sbrunner) [reply Makefile]: Thanks :-)

File: doc/developer/server_side.rst.tmpl [hunk 2]
```diff
@@ -53,18 +53,18 @@ Before running the tests, install and build all dependencies:
...
-        /app/c2cgeoportal/commons/tests
+        /opt/c2cgeoportal/commons/tests
     docker-compose exec tests pytest --verbose --color=yes \
-        /app/c2cgeoportal/geoportal/tests
+        /opt/c2cgeoportal/geoportal/tests
     docker-compose exec tests pytest --verbose --color=yes \
-        /app/c2cgeoportal/admin/tests
+        /opt/c2cgeoportal/admin/tests
     docker-compose exec qgisserver-tests pytest --verbose --color=yes \
         /src/tests/functional
```
Comment (arnaud-morvan) [inline doc/developer/server_side.rst.tmpl]: Why not some test targets ? For example: 
```
.PHONY: tests
tests: test_commons  test_geoportal test_admin
```
or something similar.

File: ci/create-new-project [hunk 3]
```diff
@@ -1,14 +1,10 @@
...
-    --build-arg=MAJOR_VERSION=${MAJOR_VERSION} --build-arg=VERSION=${TAG} .
-docker build --target=runner --tag=camptocamp/geomapfish:${TAG} \
-    --build-arg=MAJOR_VERSION=${MAJOR_VERSION} --build-arg=VERSION=${TAG} .
-docker build --target=upgrader --tag=camptocamp/geomapfish-scaffolds:${TAG} \
-    --build-arg=MAJOR_VERSION=${MAJOR_VERSION} --build-arg=VERSION=${TAG} .
-docker build --tag=camptocamp/geomapfish-config-build:${TAG} \
-    --build-arg=VERSION=${TAG} docker/config
+VERSION=${TAG} DOCKER_TAG=${TAG} make build-tools
+VERSION=${TAG} DOCKER_TAG=${TAG} make build-runner
+VERSION=${TAG} DOCKER_TAG=${TAG} make build-config-build
```
Comment (arnaud-morvan) [inline ci/create-new-project]: Not already done in ci ?

File: doc/developer/server_side.rst.tmpl (hunk 2 above)
Comment (sbrunner) [reply doc/developer/server_side.rst.tmpl]: +1 in an next pull request :-)

File: ci/create-new-project (hunk 3 above)
Comment (sbrunner) [reply ci/create-new-project]: No because TAG is a calculated value (from MAJOR_VERSION, branch name and git tag)

**Comment by arnaud-morvan** [comment]
Whaou : after some manual alembic commands from the tests container, all tests pass on my workstation !!!

**Comment by sbrunner** [comment]
> Why not run alembic from the tests container instead of having a duplicated schema as a sql file in db container ?

+1 I will do that in another pull request :-)

File: ci/create-new-project (hunk 3 above)
Comment (sbrunner) [reply ci/create-new-project]: No, because it's dynamically generated ...

**Pull request merged**